src/components/v5/common/ActionSidebar/partials/forms/ArbitraryTxsForm/partials/AddTransactionModal/JsonAbiInput.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useEffect } from 'react';
+import { useFormContext } from 'react-hook-form';
+
+import SpinnerLoader from '~shared/Preloaders/SpinnerLoader.tsx';
+import { formatText } from '~utils/intl.ts';
+import FormTextareaBase from '~v5/common/Fields/TextareaBase/FormTextareaBase.tsx';
+import Button from '~v5/shared/Button/Button.tsx';
+
+import { validateJsonAbi } from './consts.ts';
+import { MSG } from './translation.ts';
+
+interface JsonAbiInputProps {
+  loading?: boolean;
+  toggleJsonFormat: () => void;
+  showEditWarning?: boolean;
+  isFormatted: boolean;
+  toggleShowEditWarningOff: () => void;
+}
+export const JsonAbiInput: React.FC<JsonAbiInputProps> = ({
+  loading,
+  toggleJsonFormat,
+  showEditWarning,
+  isFormatted,
+  toggleShowEditWarningOff,
+}) => {
+  const { watch, clearErrors } = useFormContext();
+  const jsonAbiField = watch('jsonAbi');
+
+  useEffect(() => {
+    if (!jsonAbiField) {
+      clearErrors();
+    }
+  }, [jsonAbiField, clearErrors]);
+
+  const getTextareaOverlay = () => {
+    if (loading) {
+      return (
+        <div className="absolute bottom-0 left-0 right-0 top-0 flex items-center justify-center rounded bg-base-sprite">
+          <SpinnerLoader appearance={{ size: 'medium' }} />
+        </div>
+      );
+    }
+    if (jsonAbiField && showEditWarning) {
+      return (
+        <div className="absolute bottom-0 left-0 right-0 top-0 flex cursor-pointer items-center justify-center rounded bg-base-sprite opacity-0 transition-opacity hover:opacity-100">
+          <Button
+            onClick={toggleShowEditWarningOff}
+            className="min-w-[12.5rem]"
```

Comment:
@Nortsova as a minor improvement to correctly match `Figma` please add here `hover:border-gray-300` - but this can be done also in a subsequent PR 🫶